nixos/modules/services/web-apps/chasemapper.nix

```
@@ -0,0 +1,496 @@
+            default = 300;
+            description = ''
+              Number of bearings to store
+            '';
+            type = lib.types.int;
+          };
+          max_bearing_age = lib.mkOption {
+            default = 10;
+            description = ''
+              Maximum age of bearings, in _minutes_
+            '';
+            type = lib.types.int;
+          };
+          car_speed_gate = lib.mkOption {
+            default = 10;
+            description = ''
+              Car heading speed gate
+            '';
+            type = lib.types.int;
+          };
+          turn_rate_threshold = lib.mkOption {
+            default = 4.0;
+            description = ''
+              Turn rate threshold
+            '';
+            type = lib.types.float;
+          };
+          bearing_length = lib.mkOption {
+            default = 10;
+            description = ''
+              Bearing length in km
+            '';
+            type = lib.types.int;
+          };
+          bearing_weight = lib.mkOption {
+            default = 1.0;
+            description = ''
+              Weight of the bearing lines, in pixels
+            '';
+            type = lib.types.float;
+          };
+          bearing_color = lib.mkOption {
+            default = "red";
+            description = ''
+              Color of the bearings
+            '';
+            type = lib.types.str;
+          };
+          bearing_custom_color = lib.mkOption {
+            default = "#FF0000";
+            description = ''
+              Custom bearing color, in hexadecimal #RRGGBB
+            '';
+            type = lib.types.str;
+          };
+        };
+
+        units = {
+          unitselection = lib.mkOption {
+            default = "metric";
+            description = ''
+              unitselection allows choice of metric
+            '';
+            type = lib.types.str;
+          };
+          switch_miles_feet = lib.mkOption {
+            default = 400;
+            description = ''
+              This is the threshold for switching from miles to feet, set in metres.
+            '';
+            type = lib.types.int;
+          };
+        };
+
+        history = {
+          reload_last_position = lib.mkOption {
+            default = false;
+            description = ''
+              Enable load of last position from log files
+            '';
+            type = lib.types.bool;
+          };
+        };
+      };
+
+      openPorts = lib.mkOption {
+        type = lib.types.bool;
+        default = false;
+        description = "Whether to open firewall ports for chasemapper";
+      };
+
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    systemd.services.chasemapper = {
+      serviceConfig = {
+        Type = "simple";
+        Restart = "always";
+
+        DynamicUser = true;
+        PrivateTmp = true;
+        WorkingDirectory = cfg.package;
+        ExecStart = "${exe} --nolog -c ${chaseCfg}";
```

```suggestion
        ExecStart = "${exe} --nolog -c ${chaseCfg}";
```

Nit: why creating a variable when `exe` is only used once? Do you really think this is necessary? Why not using the expression instead?

---

I did it to make the line length a little saner but I can revert

---

Yes please.